Mathlib/GroupTheory/Coxeter/Basic.lean

```diff
@@ -74,390 +67,138 @@ presentation corresponding to a Coxeter matrix which is registered in a Coxeter
 
 ## TODO
 
-* The canonical map from the type to the Coxeter group `W` is an injection.
+* The simple reflections of a Coxeter system are distinct.
 * A group `W` registered in a Coxeter system is a Coxeter group.
 * A Coxeter group is an instance of `IsCoxeterGroup`.
 
 ## Tags
 
 coxeter system, coxeter group
--/
-
 
-universe u
+-/
 
 noncomputable section
 
-variable {B : Type*} [DecidableEq B]
-variable (M : Matrix B B ℕ)
+open Matrix Function List
 
-/-- A matrix `IsCoxeter` if it is a symmetric matrix with diagonal entries equal to one
-and off-diagonal entries distinct from one. -/
-structure Matrix.IsCoxeter : Prop where
-  symmetric : M.IsSymm := by aesop
-  diagonal : ∀ b : B, M b b  = 1 := by aesop
-  off_diagonal : ∀ b₁ b₂ : B, b₁ ≠ b₂ → M b₁ b₂ ≠ 1 := by aesop
-
-namespace CoxeterGroup
-
-namespace Relations
+variable {B B' : Type*} (M : Matrix B B ℕ) (e : B ≃ B')
 
 /-- The relations corresponding to a Coxeter matrix. -/
-def ofMatrix : B × B → FreeGroup B :=
- Function.uncurry fun b₁ b₂ => (FreeGroup.of b₁ * FreeGroup.of b₂) ^ M b₁ b₂
+def Matrix.coxeterRelation (i i' : B) : FreeGroup B :=
+ (FreeGroup.of i * FreeGroup.of i') ^ M i i'
 
 /-- The set of relations corresponding to a Coxeter matrix. -/
-def toSet : Set (FreeGroup B) :=
-  Set.range <| ofMatrix M
+def Matrix.coxeterRelationsSet : Set (FreeGroup B) :=
+  Set.range <| uncurry M.coxeterRelation
+
+/-- The proposition that `M` is a symmetric matrix with diagonal entries equal to one
+and off-diagonal entries distinct from one. -/
+structure Matrix.IsCoxeter : Prop where
+  symmetric : M.IsSymm := by aesop
+  diagonal : ∀ i : B, M i i = 1 := by aesop
+  off_diagonal : ∀ i i' : B, i ≠ i' → M i i' ≠ 1 := by aesop
 
-end Relations
+protected theorem Matrix.IsCoxeter.reindex {M} (hM : M.IsCoxeter) : (reindex e e M).IsCoxeter where
+  symmetric := by dsimp only [IsSymm]; rw [transpose_reindex, hM.symmetric]
+  diagonal := by intro b; dsimp [reindex]; exact hM.diagonal (e.symm b)
+  off_diagonal := by intro i i' hii'; dsimp [reindex]; apply hM.off_diagonal; aesop
 
-end CoxeterGroup
+theorem Matrix.reindex_isCoxeter_iff : (reindex e e M).IsCoxeter ↔ M.IsCoxeter := by
+  constructor
+  · intro h
+    simpa using h.reindex e.symm
+  · exact IsCoxeter.reindex _
 
 /-- The group presentation corresponding to a Coxeter matrix. -/
-def Matrix.CoxeterGroup := PresentedGroup <| CoxeterGroup.Relations.toSet M
+def Matrix.coxeterGroup := PresentedGroup M.coxeterRelationsSet
```

Comment:
I think ```Coxeter``` should stay capitalized since it is someone's name.  If this goes against naming conventions (I haven't looked at the rules to refresh my memory) I would argue that if this should be changed! The greats that came before us deserve this respect IMHO! :)

This goes for all other places in the code where ```Coxeter``` shows up.

Reply:
I think the naming convention is that non-prop types are lower camel case.

Reply:
Yeah, I still think for names of people we should make an exception to this.

Reply:
The old name `Matrix.CoxeterGroup` caused me genuine confusion on multiple occasions. The basic reason is this. There are already a lot of things in mathlib that have names ending in `Group` (e.g. `AddGroup` and `CommGroup`). Unlike `CoxeterGroup`, however, these are typeclasses, so they behave completely differently. This problem was exacerbated by the fact that there was also a largely unrelated namespace called `CoxeterGroup`.

Calling it `Matrix.coxeterGroup` instead of `Matrix.CoxeterGroup` gets around these issues to a large extent.

Reply:
(Also, a great mathematician once told me that using someone's name in lowercase is, in fact, a higher sign of respect than capitalizing it. I agree. When someone refers to an "abelian group", they aren't doing so to besmirch Abel's good name; they're doing so because the concept that we've rightly named after him is so fundamental and ingrained that "abelian" is just like any other common English word.)